qiskit/tools/visualization/_circuit_visualization.py

```diff
@@ -338,11 +337,9 @@ def _text_circuit_drawer(circuit, filename=None,
     Returns:
         TextDrawing: An instances that, when printed, draws the circuit in ascii art.
     """
-    basis = ("id,u0,u1,u2,u3,x,y,z,h,s,sdg,t,tdg,rx,ry,rz,"
-             "cx,cy,cz,ch,crz,cu1,cu3,swap,ccx,cswap")
-    dag_circuit = DAGCircuit.fromQuantumCircuit(circuit, expand_gates=False)
-    json_circuit = transpile_dag(dag_circuit, basis_gates=basis, format='json')
-    text_drawing = _text.TextDrawing(json_circuit, reversebits=reversebits)
+    dag = DAGCircuit.fromQuantumCircuit(circuit, expand_gates=False)
```

Comment:
what I don't get is why cant we make get_instructions use the circuit. This way a dag is never made in visualizations and it decouples the transpolar (internals from the circuit)

Simplest (not recommended) is put this line in get_instructions and only have dag inside that functions. 

Better method rewrite get_instructions to go through the circuit and make the list of ops.

Reply:
actually why is ops not just circuit._data and same for qregs and cregs

Reply:
I agree with this too, it felt a bit odd to do the circuit->dag conversion outside of the helper function, especially since this would be the only thing we use the dag for in the drawers. I've pushed up https://github.com/Qiskit/qiskit-terra/pull/1316 to switch _get_instructions() to take a circuit and just call DAGCircuit.fromQuantumCircuit() inside. I think this will be a cleaner interface moving forward.

Reply:
I don't fully disagree, but I thought the idea is to have the dag as the internal representation of the circuit. Maybe we can discuss it in an issue while moving forward with this PR?

Reply:
If we are going to do #1163 (merge circuit and dag, keeping the underlying datastructure a graph, then I think it makes sense to get the order from dag).

You may say this is premature right now, and adds extra complexity of circuit->dag->draw, but that's a different question. Where do we want to end up?

Reply:
I don't think the disagreement is on using the dag in `_utils._get_instructions()` I think that's fine as an implementation. Where I have the issue right now (and my 2 PRs #1276 and #1278  have this too) is that between the 3 backends we do circuit->dag in each of them. That's the only place where the dag is used in the circuit drawers and I feel like it's an internal detail. Take a look at #1316 it just moves the `DAGCircuit.fromQuantumCircuit()` inside the helper function so the circuit drawers only ever deal with a circuit. So instead of this line we just call `_get_instructions(circuit, reversebits=reversebits)`

Reply:
I agree with @mtreinish it’s about what is the internal api I approve @mtreinish Pr and then I am good with this if we update to that.  It’s a good job in the visualization